src/cache/surface_albedo.jl

```diff
@@ -0,0 +1,204 @@
+    return RegressionFunctionAlbedo(n, n0, p, q_clear, q_cloud, wave_slope)
+end
+
+"""
+    set_surface_albedo!(Y, p, t, α_model::ConstantAlbedo)
+
+Set the surface albedo to a constant value.
+"""
+function set_surface_albedo!(Y, p, t, α_model::ConstantAlbedo{FT}) where {FT}
+
+    (; direct_sw_surface_albedo, diffuse_sw_surface_albedo) =
+        p.radiation.radiation_model
+
+    @. direct_sw_surface_albedo = α_model.α
+    @. diffuse_sw_surface_albedo = α_model.α
+end
+
+"""
+    set_surface_albedo!(Y, p, t, α_model::RegressionFunctionAlbedo{FT})
+
+Wrapper to call the formulations of Jin et al. (2011) for direct and diffuse surface albedos of the ocean.
+"""
+function set_surface_albedo!(
+    Y,
+    p,
+    t,
+    α_model::RegressionFunctionAlbedo{FT},
+) where {FT}
+
+    (; direct_sw_surface_albedo, diffuse_sw_surface_albedo, cos_zenith) =
+        p.radiation.radiation_model
+
+    λ = FT(0) # spectral wavelength (not used for now)
+    μ = cos_zenith
+
+    u_int = Fields.field_values(Fields.level(Y.c.uₕ, 1))
+    int_local_geometry_values =
+        Fields.field_values(Fields.level(Fields.local_geometry_field(Y.c), 1))
+
+    wind_speed = p.scratch.ᶠtemp_field_level
+    wind_speed_values = Fields.field_values(wind_speed)
+    @. wind_speed_values =
+        Geometry._norm.(
+            CT12.(u_int, int_local_geometry_values),
+            int_local_geometry_values,
+        )
+    # TODO: Is there a neater way of defining wind speed? E.g., do we want to use SF.jl windspeed? including gustiness?  Or do we want to extrapolate to the surface using the MOST profile? It would be good to expose SF.surface_conditions in that case.
+
+    # set the direct and diffuse surface albedos
+    direct_sw_surface_albedo .=
+        ones(axes(direct_sw_surface_albedo)) .* reshape(
+            surface_albedo_direct(
+                α_model,
+            ).(λ, μ, RRTMGPI.field2array(wind_speed)),
+            1,
+            :,
+        )
+    diffuse_sw_surface_albedo .=
+        ones(axes(direct_sw_surface_albedo)) .* reshape(
+            surface_albedo_diffuse(
+                α_model,
+            ).(λ, μ, RRTMGPI.field2array(wind_speed)),
+            1,
+            :,
+        )
+end
+
+"""
+    surface_albedo_direct(α_model::RegressionFunctionAlbedo{FT}) where {FT}
+
+Set the surface albedo using the regression function of Jin et al. (2011).
+"""
+function surface_albedo_direct(α_model::RegressionFunctionAlbedo{FT}) where {FT}
+    α_dir =
+        (λ, cosθ, u) -> begin
+            # relative refractive index of water and air (n = n_w/n_a)
+            n = α_model.n
+
+            # refractive index of water for visible light
+            n0 = α_model.n0
+
+            # mean wave slope distribution width
+            σ = α_model.wave_slope(u)
+
+            # Fresnel reflectance (assuming equal contribution of the p-polorized and s-polarized components, and using the perfect dielectric medium)
+            sin(cosθ) = sqrt(1 - cosθ^2)
```

Comment:
```suggestion
            sinθ(cosθ) = sqrt(1 - cosθ^2)
```
I think this is more consistent with the doc and less confusing, as sin(cosθ) makes me think it's the sin of cosθ when I look at it first time. But it's up to you.

Reply:
No that makes sense, thanks!